### ISSUE TITLE
Crash 2022-12-06.8d36963 macos

### ISSUE THREAD

**Issue by magnusviri** [freeorion/freeorion#4344]
Bug Report
==========

When loading this save game, FreeOrion 2022-12-06.8d36963 crashes. macOS 10.6, M1 ARM MacBook.

Turn 45 loads: [FreeOrion_Ah_Ah_0045_20221209_180649.sav.zip](https://github.com/freeorion/freeorion/files/10198926/FreeOrion_Ah_Ah_0045_20221209_180649.sav.zip)

Turn 46 crashes [FreeOrion_Ah_Ah_0046_20221209_180232.sav.zip](https://github.com/freeorion/freeorion/files/10198921/FreeOrion_Ah_Ah_0046_20221209_180232.sav.zip)

[Crash Report](https://github.com/freeorion/freeorion/files/10198922/Crash.txt)

**Comment by magnusviri** [comment]
I think I know why turn 46 crashes! On turn 45 I ignored the sensor ghost Space Cloud headed to Dunatis. If I don't ignore the sensor ghost then it doesn't crash.

**Comment by o01eg** [comment]
Stacktrace:

```
(gdb) bt
#0  0x00007ffff78ad5f1 in Universe::ForgetKnownObject (this=0x7fffffffd248, empire_id=1, object_id=1968) at /home/oleg/devel/freeorion/universe/Universe.cpp:1979
#1  0x00007ffff79c5c02 in OrderSet::ApplyOrders (this=0x7fffffffdc00, context=...) at /usr/include/c++/12/bits/shared_ptr_base.h:1665
#2  0x00005555559a63c8 in MapWnd::InitTurn (this=0x55555698ff60, context=...) at /home/oleg/devel/freeorion/UI/MapWnd.cpp:2865
#3  0x0000555555703cd1 in PlayingTurn::PlayingTurn (this=0x555556df3af0, ctx=...) at /usr/include/c++/12/bits/shared_ptr_base.h:1665
#4  0x000055555570cef4 in boost::statechart::state<PlayingTurn, PlayingGame, boost::mpl::list<mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, (boost::statechart::history_mode)0>::shallow_construct (pContext=..., 
    outermostContextBase=...) at /usr/include/boost/statechart/detail/counted_base.hpp:72
(gdb) p obj
$2 = (const UniverseObject * const) 0x7fff902622d0
(gdb) p *obj
$3 = {<std::enable_shared_from_this<UniverseObject>> = <invalid address>, _vptr.UniverseObject = 0x5552a96c75a2, 
  StateChangedSignal = {<boost::signals2::signal_base> = {<boost::noncopyable_::noncopyable> = {<boost::noncopyable_::base_token> = {<No data fields>}, <No data fields>}, 
      _vptr.signal_base = 0x1674cfd22284df2c}, <boost::signals2::detail::std_functional_base<>> = {<No data fields>}, _pimpl = {px = 0x7fff902623f0, pn = {pi_ = 0x7fff90262570}}}, 
  static INVALID_POSITION = -100000, static INVALID_OBJECT_AGE = -1073741825, static SINCE_BEFORE_TIME_AGE = 1073741825, m_name = "Monsters", m_system_id = -1, m_id = 1968, 
  m_owner_empire_id = -1, m_created_on_turn = -32768, m_x = 809.88753952161153, m_y = 517.03439914211822, m_meters = {m_flat_tree = {
      m_data = {<boost::container::dtl::flat_tree_value_compare<std::less<MeterType>, boost::container::dtl::pair<MeterType, Meter>, boost::container::dtl::select1st<MeterType> >> = {<std::less<MeterType>> = {<std::binary_function<MeterType, MeterType, bool>> = {<No data fields>}, <No data fields>}, <No data fields>}, m_seq = {
          m_holder = {<boost::container::new_allocator<boost::container::dtl::pair<MeterType, Meter> >> = {<No data fields>}, m_start = 0x7fff90262590, m_size = 1, m_capacity = 1}}}, 
      static has_stored_allocator_type = true}}, m_specials = {m_flat_tree = {
      m_data = {<boost::container::dtl::flat_tree_value_compare<std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, boost::container::dtl::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::pair<int, float> >, boost::container::dtl::select1st<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >> = {<std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >> = {<std::binary_function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool>> = {<No data fields>}, <No data fields>}, <No data fields>}, m_seq = {
          m_holder = {<boost::container::new_allocator<boost::container::dtl::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::pair<int, float> > >> = {<No data fields>}, m_start = 0x0, m_size = 0, m_capacity = 0}}}, static has_stored_allocator_type = true}}, m_type = UniverseObjectType::OBJ_FLEET}
```

**Comment by Vezzra** [comment]
@geoffthemedio, @o01eg, is this a macOS specific issue or can you reproduce it on Windows and/or Linux?

**Comment by o01eg** [comment]
I reproduced it on Linux.

**Comment by geoffthemedio** [comment]
I'm unable to reproduce a crash as described, but I think I found a plausible cause and fix in #4347

@o01eg a test with that pull request would be helpful...

**Comment by o01eg** [comment]
@geoffthemedio PR fixes crash.
